src/main/java/org/testng/Assert.java

```diff
@@ -1077,7 +1076,24 @@ public static void assertEquals(Object[] actual, Object[] expected, String messa
         fail("Arrays not equal: " + Arrays.toString(expected) + " and " + Arrays.toString(actual));
       }
     }
-    assertEquals(Arrays.asList(actual), Arrays.asList(expected), message);
+    if (actual.length != expected.length) {
```

Comment:
What if we add a `public static void assertEquals(Object[][] actual, Object[][] expected, String message)`?
Does it work?

Reply:
No. Its not working this way.